.github/workflows/pr-build.yml

```diff
@@ -1,45 +1,122 @@
+        
 jobs:
   build:
-    name: Build
+    name: Build the application
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pull-requests: write
     steps:
+      - name: Setup
+        env: 
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          gh repo clone ${{ github.repository }}
+          cd revanced-manager
+          gh repo set-default ${{ github.repository }}
+          gh pr checkout ${{ inputs.pr-number }}
+          
+          echo "DATETIME=$( TZ='UTC+0' date --rfc-email )" >> $GITHUB_ENV
+          echo "COMMIT_HASH=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+          
       - name: Checkout
         uses: actions/checkout@v4
         with:
-          # Make sure the release step uses its own credentials:
-          # https://github.com/cycjimmy/semantic-release-action#private-packages
           persist-credentials: false
-          fetch-depth: 0
+          
       - name: Setup JDK
         uses: actions/setup-java@v3
         with:
           java-version: '11'
           distribution: 'zulu'
+          cache: gradle
+          
       - name: Setup Flutter
         uses: subosito/flutter-action@v2
         with:
-          channel: 'stable'
-          cache: true
+          channel: ${{ inputs.flutter-branch }}
+          cache: ${{ inputs.flutter-cache }}
+          
       - name: Install Flutter dependencies
         run: flutter pub get
+        
       - name: Generate files with Builder
         run: flutter packages pub run build_runner build --delete-conflicting-outputs
+        
       - name: Build with Flutter
-        env:
+        continue-on-error: true
+        id: flutter-build
+        env: 
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: flutter build apk --debug
+        run: |
+          flutter build apk --${{ inputs.app-flavour }};
+          
+      - name: Prepare to comment
+        run: |
+          if [[ "${{ steps.flutter-build.outcome }}" == "success" ]]; then
+            echo "MESSAGE=✅ ReVanced Manager ${{ env.COMMIT_HASH }} build succeeded." >> $GITHUB_ENV
+          else
+            echo "MESSAGE=🚫 ReVanced Manager ${{ env.COMMIT_HASH }} build failed." >> $GITHUB_ENV
+          fi
+          
+      - name: "Comment to Pull Request #${{ inputs.pr-number }}"
+        uses: thollander/actions-comment-pull-request@v2
+        with:
+          GITHUB_TOKEN: ${{ github.token }}
+          pr_number: ${{ inputs.pr-number }}
+          mode: recreate
+          message: |
+            ## ⚒️ ReVanced PR Build workflow
+            
+            ${{ env.MESSAGE }}
+            
+            You can see more details in run [${{ github.run_id }}](https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }})!
+
+            ### ⚙️ Overview
+            - App flavor: ${{ inputs.app-flavour }}
+            - Flutter branch: ${{ inputs.flutter-branch }}
+            - Flutter cache: ${{ inputs.flutter-cache }}
```

Comment:
```suggestion
            - Branch: ${{ inputs.flutter-branch }}
            - Cache: ${{ inputs.flutter-cache }}
```

Reply:
In my opinion, the word "Branch" & "Cache" might be too ambiguous:

* *Are we building ReVanced Manager at the `beta` branch or using the Flutter beta branch to build ReVanced*

* *Are we caching both Gradle and Flutter or just Flutter?*

Reply:
> Are we building ReVanced Manager at the beta branch or using the Flutter beta branch to build ReVanced Manager?

This question should be treated as redundant as this branch is not aware of the branch that will eventually feature the compose version

Reply:
Resolved @ 5068f37